index.js

```diff
@@ -0,0 +1,95 @@
+var express = require('express')
+var bodyParser = require('body-parser')
+var models = require('./models')
+
+var app = express()
+
+app.get('/movies', async (request, response) => {
+    const matchingmovies = await models.movies.findAll({
+        include: [{ model: models.directors }, { model: models.genres }]
+    })
+    return response.send(matchingmovies)
+})
+
+app.get('/movies/:identifier', async (request, response) => {
+    const movies = await models.movies.findAll({
+        include: [{
+            all: true
+        }],
+        where: {
+            id: request.params.identifier
+        }
+    })
+    response.send(movies)
+})
+
+app.get('/directors/:identifier', async (request, response) => {
+    const director = await models.directors.findAll({
+        include: {
+            model: models.movies
+        },
+        where: {
+            id: request.params.identifier
+        }
+    })
+    response.send(director)
+})
+
+app.get('/genres/:identifier', async (request, response) => {
+    const genre = await models.genres.findAll({
+        include: {
+            model: models.movies
+        },
+        where: {
+            genre: request.params.identifier
+        }
+    })
+    response.send(genre)
+})
+
+app.post('/movies', bodyParser.json(), async (request, response) => {
+    const { title, directors, releaseDate, rating, runTime, genres } = request.body
+    const names = directors.split(', ')
+    const genresArray = genres.split(', ')
+
+    if (!title || !directors || !releaseDate || !rating || !runTime || !genres) {
+        response.status(400).send('The following attributes are required: title, directors, releaseDate, rating, runTime, genres')
+    } else {
+        const newMovie = await models.movies.create({ title, releaseDate, rating, runTime })
+        response.status(201).send(newMovie)
```

Comment:
By responding here you're sending back a 201 with just the movie data to the user. This means if the other data creates fail the user will never know since they've already received a 201 response. It also means they are receiving back just the movie data and not the movie along with directors and genres.